File: src/com/fabrica/muebles/modelo/cliente.py
from datetime import date


class Cliente:
    """
    Clase modelo que representa un Cliente en el sistema
    Corresponde a la tabla: clientes

    Autor: William Alonso Samaca Lopez
    Sistema de Gestión Fábrica de Muebles
    """

    def __init__(self, id_cliente=None, nombre=None, telefono=None, direccion=None,
                 email=None, codigo_mueble=None, tipo_documento='CC', numero_documento=None,
                 tipo_mueble_vendido=None, cantidad=1, valor_mueble=0, ruta_foto=None,
                 fecha_registro=None):
        """
        Constructor de la clase Cliente

        Args:
            id_cliente: ID del cliente (None para nuevos clientes)
            nombre: Nombre completo del cliente
            telefono: Teléfono del cliente
            direccion: Dirección del cliente
            email: Email del cliente
            codigo_mueble: Código del mueble vendido (asignado por producción)
            tipo_documento: Tipo de documento (CC o NIT)
            numero_documento: Número de documento
            tipo_mueble_vendido: Tipo de mueble que se le vendió
            cantidad: Cantidad de muebles vendidos
            valor_mueble: Valor/precio del mueble
            ruta_foto: Ruta de la foto del mueble
            fecha_registro: Fecha de registro (None para fecha actual)
        """
        self.id_cliente = id_cliente
        self.nombre = nombre
        self.telefono = telefono
        self.direccion = direccion
        self.email = email
        self.codigo_mueble = codigo_mueble
        self.tipo_documento = tipo_documento
        self.numero_documento = numero_documento
        self.tipo_mueble_vendido = tipo_mueble_vendido
        self.cantidad = cantidad
        self.valor_mueble = valor_mueble
        self.ruta_foto = ruta_foto
        self.fecha_registro = fecha_registro or date.today()

    def __str__(self):
        """Representación en string del objeto Cliente"""
        return (f"Cliente{{codigo_mueble='{self.codigo_mueble}', nombre='{self.nombre}', "
                f"tipo_doc='{self.tipo_documento}', num_doc='{self.numero_documento}', "
                f"telefono='{self.telefono}', email='{self.email}'}}")

    def __repr__(self):
        return self.__str__()

File: modulo-java/src/com/fabrica/muebles/modelo/Cliente.java
package com.fabrica.muebles.modelo;

import java.sql.Date;

/**
 * Clase modelo que representa un Cliente en el sistema
 * Corresponde a la tabla: clientes
 * 
 * Autor: William Alonso Samaca Lopez
 * Sistema de Gestión Fábrica de Muebles
 */
public class Cliente {
    
    private int idCliente;
    private String nombre;
    private String telefono;
    private String direccion;
    private String email;
    private Date fechaRegistro;
    private int estado;  // 1 = Activo, 0 = Inactivo
    
    public Cliente() {
    }
    
    public Cliente(String nombre, String telefono, String direccion, String email) {
        this.nombre = nombre;
        this.telefono = telefono;
        this.direccion = direccion;
        this.email = email;
        this.fechaRegistro = new Date(System.currentTimeMillis());
        this.estado = 1;
    }
    
    public Cliente(int idCliente, String nombre, String telefono, String direccion, 
                   String email, Date fechaRegistro, int estado) {
        this.idCliente = idCliente;
        this.nombre = nombre;
        this.telefono = telefono;
        this.direccion = direccion;
        this.email = email;
        this.fechaRegistro = fechaRegistro;
        this.estado = estado;
    }
    
    public int getIdCliente() {
        return idCliente;
    }
    
    public void setIdCliente(int idCliente) {
        this.idCliente = idCliente;
    }
    
    public String getNombre() {
        return nombre;
    }
    
    public void setNombre(String nombre) {
        this.nombre = nombre;
    }
    
    public String getTelefono() {
        return telefono;
    }
    
    public void setTelefono(String telefono) {
        this.telefono = telefono;
    }
    
    public String getDireccion() {
        return direccion;
    }
    
    public void setDireccion(String direccion) {
        this.direccion = direccion;
    }
    
    public String getEmail() {
        return email;
    }
    
    public void setEmail(String email) {
        this.email = email;
    }
    
    public Date getFechaRegistro() {
        return fechaRegistro;
    }
    
    public void setFechaRegistro(Date fechaRegistro) {
        this.fechaRegistro = fechaRegistro;
    }
    
    public int getEstado() {
        return estado;
    }
    
    public void setEstado(int estado) {
        this.estado = estado;
    }
    
    public String getEstadoTexto() {
        return (estado == 1) ? "Activo" : "Inactivo";
    }
    
    @Override
    public String toString() {
        return "Cliente{" +
                "ID=" + idCliente +
                ", nombre='" + nombre + '\'' +
                ", telefono='" + telefono + '\'' +
                ", email='" + email + '\'' +
                ", estado='" + getEstadoTexto() + '\'' +
                '}';
    }
}